Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -15,6 +15,14 @@ public final class MapboxMap {
         }
     }
 
+    private var eventHandlers = WeakSet<MapEventHandler>()
+
+    deinit {
+        eventHandlers.allObjects.forEach {
+            $0.cancel()
+        }
+    }
+
```

Comment:
Do we need a test for this?

Reply:
There is..sort of. See [below](https://github.com/mapbox/mapbox-maps-ios/pull/339/files#r631080035).